File: lib/models/order.dart
class OrderModel {
  final String id;
  final String partCode;
  final String partName;
  final int quantity;
  final double price;
  final double gst;
  final double totalAmount;
  final String status;
  final DateTime createdAt;

  OrderModel({
    required this.id,
    required this.partCode,
    required this.partName,
    required this.quantity,
    required this.price,
    required this.gst,
    required this.totalAmount,
    required this.status,
    required this.createdAt,
  });

  factory OrderModel.fromJson(Map<String, dynamic> json) {
    return OrderModel(
      id: json['id'],
      partCode: json['part_code'],
      partName: json['part_name'],
      quantity: json['quantity'],
      price: double.parse(json['price'].toString()),
      gst: double.parse(json['gst'].toString()),
      totalAmount: double.parse(json['total_amount'].toString()),
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: backend/models/Order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
  part_code: { type: String, required: true },
  part_name: { type: String, required: true },
  quantity: { type: Number, required: true },
  price: { type: Number, required: true },
  gst: { type: Number, required: true },
  total_amount: { type: Number, required: true },
  status: { type: String, default: 'Confirmed' }
}, {
  timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
});

// Map _id to id virtual and serialize properly
orderSchema.virtual('id').get(function() {
  return this._id.toHexString();
});

orderSchema.set('toJSON', {
  virtuals: true,
  transform: function(doc, ret) {
    ret.id = ret._id.toString();
    delete ret._id;
    delete ret.__v;
    return ret;
  }
});

module.exports = mongoose.models.Order || mongoose.model('Order', orderSchema);
